docs/addons.md

```diff
@@ -0,0 +1,167 @@
+```
+
+You can check which addons are currently enabled with the command:
+
+```bash
+microk8s.status
+```
+
+This also conveniently lists all the known addons, cutting down on trips to
+these docs! The addons currently available are listed below, and explained in
+more detail in the sections following.
+
+**[dashboard][]**: The standard Kubernetes Dashboard.
+
+**[dns][]**: Deploys CoreDNS. This add-on may be required by others - it is
+recommended you always enable it. In restricted environments you may need to
+update the upstream DNS servers ([see below][dns]).
+
+**[fluentd][]**: Deploy the Elasticsearch-Fluentd-Kibana logging and monitoring
+solution.
+
+**[gpu][]**:  Enable support for GPU accelerated workloads using the nVidia runtime.
+
+**[ingress][]**: A simple ingress controller for external access.
+
+**[istio][]**: Adds the core [Istio][istio-docs] services.
+
+**[jaeger][]**: Deploy the [Jaeger Operator][jaeger-docs] in the “simplest” configuration.
+
+**[knative][]**: Adds the [Knative][knative-docs] middleware to your cluster.
+
+**[linkerd][]**: Deploys the linkerd service mesh. [See below](#linkerd) for configuration.
+
+**[metrics-server][]**: Adds the Kubernetes Metrics Server for API access to service metrics.
+
+**[prometheus]()**: Deploys the [Prometheus Operator][prometheus-docs].
+
+**[rbac]()**: Enable Role Based Access Control for authorisation. Note that this is incompatible with some other add-ons ([See notes](#rbac))
+
+**[registry][]**: Deploy a private image registry and expose it on localhost:32000. The storage add-on will be enabled as part of this add-on. See the registry documentation for more details.
+
+**[storage][]**: Create a default storage class which allocates storage from a host directory.
+
+
+### dns
+
+As noted above, this deploys CoreDNS to supply address resolution services to Kubernetes. This service is commonly required by other addons, so it is recommended that you enable it.
+
+By default it points to Google's 8.8.8.8, 8.8.4.4 servers for resolving
+addresses. This can be changed by running the command:
+
+```bash
+microk8s.kubectl -n kube-system edit configmap/coredns
+```
+
+This will invoke the `vim` editor so you can alter the configuration.
+
+### dashboard
+
+The standard Kubernetes Dashboard is a convenient way to keep track of the
+activity and resource use of MicroK8s
+
+
+To log in to the Dashboard, you will need the access token. This is generated randomly on deployment, so a few commands are needed to retrieve it:
+
+```bash
+token=$(microk8s.kubectl -n kube-system get secret | grep default-token | cut -d " " -f1)
+microk8s.kubectl -n kube-system describe secret $token
+```
+
+Next, you need a connection to the API server. While the MicroK8s snap will have an IP address on your local network, the recommended way to do this is through the proxy service. You can initiate the proxy with the command:
+
+```bash
+microk8s.kubectl proxy --accept-hosts=.* --address=0.0.0.0 &
```

Comment:
I am assuming that running proxy is the 'approved' method

Reply:
Yes this is the recommended way based on the upstream docs https://kubernetes.io/docs/tasks/access-application-cluster/web-ui-dashboard/#accessing-the-dashboard-ui

Note that this process is slightly different when RBAC is enabled. What we currently do is print the following message when the user does a `microk8s.enable dashboard`:
```
If RBAC is not enabled access the dashboard using the default token retrieved with:

token=$(microk8s.kubectl -n kube-system get secret | grep default-token | cut -d " " -f1)
microk8s.kubectl -n kube-system describe secret $token

In an RBAC enabled setup (microk8s.enable RBAC) you need to create a user with restricted
permissions as shown in https://github.com/kubernetes/dashboard/wiki/Creating-sample-user
```

We also got this issue on improving the UX: https://github.com/ubuntu/microk8s/issues/584

We may end-up printing the token as well. Any suggestions are welcomed.